gocontext.go

```diff
@@ -2,31 +2,31 @@ package opentracing
 
 import "golang.org/x/net/context"
 
-type goContextKey int
+type contextKey int
 
-const activeSpanKey goContextKey = iota
+const activeSpanKey contextKey = iota
 
-// GoContextWithSpan seturns a new `context.Context` that holds a reference to
+// ContextWithSpan returns a new `context.Context` that holds a reference to
 // the given `Span`.
 //
-// NOTE: We use the term "GoContext" to minimize confusion with TraceContext.
-func GoContextWithSpan(ctx context.Context, span Span) context.Context {
-	return context.WithValue(ctx, activeSpanKey, span)
+// The second return value is simply the `span` passed in:
+// this can save some typing and is only provided as a convenience.
+func ContextWithSpan(ctx context.Context, span Span) (context.Context, Span) {
```

Comment:
since there's no longer a `TraceContext` type, we can finally refer to `GoContext` as `Context` :)